File: java/src/test/java/com/adaptionsoft/games/trivia/BufferPresenter.java
package com.adaptionsoft.games.trivia;

import com.adaptionsoft.games.uglytrivia.Presenter;

import java.util.ArrayList;
import java.util.List;

public class BufferPresenter implements Presenter {
    private final List<String> messages = new ArrayList<>();

    @Override
    public void present(String message) {
        messages.add(message);
    }

    public List<String> getBufferedMessages() {
        return messages;
    }
}

File: kotlin/src/test/kotlin/com/adaptionsoft/games/trivia/BufferPresenter.kt
package com.adaptionsoft.games.trivia

import com.adaptionsoft.games.uglytrivia.Presenter
import java.util.ArrayList

internal class BufferPresenter : Presenter {
    private val messages: MutableList<String> = ArrayList()

    override fun present(message: String) {
        messages.add(message)
    }

    val bufferedMessages: List<String>
        get() = messages
}